### ISSUE TITLE
tf.nn.embedding_lookup gradient shape is no longer fully defined in 2.5.0rc0

### ISSUE THREAD

**Issue by guillaumekln** [tensorflow/tensorflow#48430]
**System information**
- Have I written custom code: Yes
- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from: binary
- TensorFlow version: `2.5.0rc0`
- Python version: 3.6.9

**Describe the current behavior**

In TensorFlow 2.5.0rc0, the dense shape of the `tf.nn.embedding_lookup` gradient is no longer fully defined. The shape is `(None, None)` instead of the static shape of the embedding variable.

 The same code worked in previous TensorFlow versions.

**Describe the expected behavior**

The `shape` attribute of the sparse gradient should be the fully defined shape of the embedding variable.

**Standalone code to reproduce the issue**

```python
import tensorflow as tf

class MyLayer(tf.keras.layers.Layer):
    def build(self, input_shape):
        self.embedding = tf.Variable(tf.random.uniform([50, 16]))

    def call(self, x):
        return tf.nn.embedding_lookup(self.embedding, x)

layer = MyLayer()

@tf.function
def _run(x):
    with tf.GradientTape() as tape:
        y = layer(x)
        loss = tf.math.reduce_sum(y)
    gradients = tape.gradient(loss, layer.weights)
    print("Gradient shape:", gradients[0].shape)

_run(tf.random.uniform([4, 16], minval=0, maxval=50, dtype=tf.int64))
```

**Other info / logs**

Output with 2.5.0rc0 (incorrect):

```text
Gradient shape: (None, None)
```

Output with 2.4.1 (correct):

```text
Gradient shape: (50, 16)
```

**Comment by amahendrakar** [comment]
@jvishnuvardhan,
I was able to reproduce the issue. Code works fine with TF v2.4.1, whereas with TF v2.5.0-rc0 and TF-nightly the output is `None`.

Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/6e31d91ae1ddf017ea1e9f1df9d90761/48430.ipynb). Thanks!

**Comment by guillaumekln** [comment]
The regression was introduced in this commit: https://github.com/tensorflow/tensorflow/commit/7ba58c1fe7ce110dfd91127ef3c9b629a5ad5e32

**Comment by allenlavoie** [comment]
Fixed in master, still needs a cherry-pick or a rollback in the release branch (probably the former).

Thank you for testing the release and the great report!
